Applications/Calculator/Calculator.cpp

```diff
@@ -105,32 +105,30 @@ double Calculator::finish_operation(double argument)
         return argument;
 
     case Operation::Add:
-        if ((argument > INT_MAX - m_saved_argument) || (m_saved_argument > INT_MAX + argument)) {
+        if (__builtin_add_overflow_p((int)m_saved_argument, (int)argument, (int)&res)) {
```

Comment:
This line here works during testing (for example inputting "1073741824 / .5" gives an expected error) but I don't feel as if it's robust enough.